### PR TITLE
CDCINITIALIZER-183: Generate cache everytime it is not updated

### PR DESCRIPTION
**Pull request by bmmsantos**

### CODE REVIEW

File: cdc-accelerator/core/sitesCache.js [hunk 1]
```diff
@@ -40,4 +38,27 @@ export default class SitesCache {
...
     }
+
+    static isCacheUpdated(configuration) {
+        if (!configuration?.cache) {
+            return false
+        }
+        let updated = true
+        if (configuration.source) {
+            configuration.source.forEach((entry) => {
+                updated &= configuration.cache.some(({ apiKey }) => {
```
Comment (luismarques-io) [inline cdc-accelerator/core/sitesCache.js:49]: Using `&=` you are converting `updated` to a number, it's a good practice to not change the type of the variables. You can try to use `&&=` instead.

File: cdc-accelerator/core/sitesCache.js [hunk 2]
```diff
@@ -40,4 +38,27 @@ export default class SitesCache {
...
+                    return entry.apiKey === apiKey
+                })
+            })
+        }
+
+        if (updated && configuration.deploy) {
+            configuration.deploy.forEach((entry) => {
+                updated &= configuration.cache.some(({ apiKey }) => {
```
Comment (luismarques-io) [inline cdc-accelerator/core/sitesCache.js:57]: Same here (`&=`)

File: cdc-accelerator/core/sitesCache.js [hunk 3]
```diff
@@ -40,4 +38,27 @@ export default class SitesCache {
...
+                    return entry.apiKey === apiKey
+                })
+            })
+        }
+        return updated ? true : false
```
Comment (luismarques-io) [inline cdc-accelerator/core/sitesCache.js:62]: If `updated` stays a boolean, there is no need for this expression.

File: cdc-accelerator/core/sitesCache.js [hunk 4]
```diff
@@ -40,4 +38,27 @@ export default class SitesCache {
...
+            })
+        }
+
+        if (updated && configuration.deploy) {
+            configuration.deploy.forEach((entry) => {
+                updated &= configuration.cache.some(({ apiKey }) => {
+                    return entry.apiKey === apiKey
+                })
+            })
+        }
```
Comment (luismarques-io) [inline cdc-accelerator/core/sitesCache.js:61]: The code inside `if (configuration.source) {` seems to be duplicated in `if (updated && configuration.deploy) {`. We can create a separate function for that specific part.

**Review by luismarques-io — CHANGES_REQUESTED** [review]

File: cdc-accelerator/core/sitesCache.js (hunk 1 above)
Comment (bmmsantos) [reply cdc-accelerator/core/sitesCache.js]: Good catch :) changed

File: cdc-accelerator/core/sitesCache.js (hunk 2 above)
Comment (bmmsantos) [reply cdc-accelerator/core/sitesCache.js]: changed

File: cdc-accelerator/core/sitesCache.js (hunk 3 above)
Comment (bmmsantos) [reply cdc-accelerator/core/sitesCache.js]: changed

File: cdc-accelerator/core/sitesCache.js (hunk 4 above)
Comment (bmmsantos) [reply cdc-accelerator/core/sitesCache.js]: done

**Review by luismarques-io — APPROVED** [review]

**Pull request merged**